test/840_weave_kube_3_test.sh

```diff
@@ -72,17 +73,38 @@ assert "run_on $HOST2 ps aux | grep -c '[d]efunct'" "0"
 assert "run_on $HOST3 ps aux | grep -c '[d]efunct'" "0"
 
 # See if we can get some pods running that connect to the network
-run_on $HOST1 "$KUBECTL run hello --image=weaveworks/hello-world --replicas=3"
+run_on $HOST1 "$KUBECTL run nettest --image=$IMAGE --replicas=3 -- -peers=3 -dns-name=nettest.default.svc.cluster.local."
+# Create a headless service so they can be found in Kubernetes DNS
+run_on $HOST1 "$KUBECTL create -f -" <<EOF
+apiVersion: v1
+kind: Service
+metadata:
+  name: nettest
+spec:
+  clusterIP: None
+  ports:
+  - port: 80
+    targetPort: 80
```

Comment:
Yes; actually it doesn't matter because it's a headless service, but I may as well make them match.

It doesn't work at all if you leave the port out ☹️